### ISSUE TITLE
Run Bitcoin daemon using BTCPay Server

### ISSUE THREAD

**Issue by whiteyhat** [btcpayserver/btcpayserver#802]
I have seen there is a bitcoin cli in the dockerized version. Can I run set up the RPC connection and run the daemon?

When trying to change the RPC user or password I get the error code: `-32601`

Any ideas to solve this issue?

**Comment by whiteyhat** [comment]
When I try to connect I also receive this message:
```
Make sure the bitcoind server is running and that you are connecting to the correct RPC port
```

**Comment by whiteyhat** [comment]
Hello?

**Comment by Kukks** [comment]
Patience is a virtue :)

I'm not really sure what you mean. Are you trying to modify the btcpayserver-docker environment?

**Comment by whiteyhat** [comment]
I thought btcpayserver-docker provided a bitcoin daemon to run it

**Comment by Kukks** [comment]
It does, still not sure what you want to do though

**Comment by whiteyhat** [comment]
I want to run the bitcoin daemon. What is the command to do it in this dockerized version? Because when I try `bitcoind` nothing happens :)

**Comment by hubiktomas** [comment]
You can access Bitcoin Core RPC client using `bitcoin-cli.sh`

**Comment by whiteyhat** [comment]
> You can access Bitcoin Core RPC client using `bitcoin-cli.sh`

Yes @hubiktomas but how can I start the daemon?

**Comment by hubiktomas** [comment]
The daemon runs in the docker in the default setup. Or what do you mean?

**Comment by whiteyhat** [comment]
I dont meant to use the bitcoin-cli which is a command-line interface for Bitcoin Core client that connects to a running instance of bitcoind daemon.

I want to start the bitcoind which is a Bitcoin service daemon, which is a program that runs in the background and with which a user can't usually interact directly during it's runtime. This is a full node server that downloads that creates a P2P network and synchronizes the blockchain and verifies transactions and blocks. So with I can connect externally using the RFC credentials

**Comment by hubiktomas** [comment]
I know what it is. I am just saying that this Bitcoin full node runs in the docker and the command that I posted before is used to interact with this dockerized daemon. You are not starting the daemon by running bitcoind command. You are starting it by running the container. Maybe there is some misunderstanding in what you are trying to achieve.

**Comment by whiteyhat** [comment]
Oh, I see. Based on my experience trying to set up the daemon using the command `bitcoin-cli.sh` as far as I have reached trying to start the daemon is here:
```
error: Could not connect to the server 127.0.0.1:8332

Make sure the bitcoind server is running and that you are connecting to the correct RPC port
```

Do you know what command can I use with `bitcoin-cli.sh` to set up the daemon running? or if it is already running see the `bitcoin.conf` to see the RPC credentials?

**Comment by hubiktomas** [comment]
See the list of docker volumes. The full path will be there. It will be probably something like this: `/var/lib/docker/volumes/production_bitcoin_datadir/_data` But you should have a look at the docker config to see what ports are actually exposed.

**Comment by whiteyhat** [comment]
Umhhh.. nice one @hubiktomas! 👍 

 I can see the data from the `bitcoin.conf` where are the RPC credentials supposed to be? I can not find them in the `bitcoin.conf`

**Comment by whiteyhat** [comment]
I can only see:
`rpcport`
`rpcallowip`
`rpcporta`

 But nothing about the credentials or how to run the bitcoind dockerized version

**Comment by whiteyhat** [comment]
Anyone know where are the `rpcuser` and `rpcpassword`of the `/var/lib/docker/volumes/production_bitcoin_datadir/_data/bitcoin.conf` or the `bitcoin-cli.sh` to update it?

**Comment by NicolasDorier** [comment]
If you update, there should be one in `/var/lib/docker/volumes/production_bitcoin_datadir/_data/bitcoin.conf` there is a hard coded creds that I need to use for lnd.

**Comment by pavlenex** [comment]
Closing the issue as it appears that it's been resolved. If it's not, feel free to let us know.

**Comment by AfolabiOlaoluwa** [comment]
I have the same problem also, and I do not think I have to open another issue. First of all, kindly forgive me of my blockchain knowledge if this sounds noob. 

I checked through the Docker files I am running on my system which is inside the directory **BTCPayServe.Tests** file, one thing I figured out is the Bitcoin Daemon is **bitcoind** and a docker image is provided for it inside the docker-compose.yml file. Looking further, I saw bitcoind RPC credentials which you all will agree with me that if I need to run this on production, I cant expose credentials (i.e. rpcuser=ceiwHEbqWI83, rpcpassword=DwubwWsoo3) like that. See below;

```
services:

  tests:
    build:
      context: ..
      dockerfile: BTCPayServer.Tests/Dockerfile
      args:
        CONFIGURATION_NAME: Release
    environment:
      TESTS_BTCRPCCONNECTION: server=http://bitcoind:43782;ceiwHEbqWI83:DwubwWsoo3
      TESTS_BTCNBXPLORERURL: http://nbxplorer:32838/
```

Then I thought of changing those credentials to have my own environment because I will be running it on my own private host or server. This now brought me to the question that opened this issue. 

From the comments I have read so far from this closed issue, it appears that for the production environment, I can set all my personal credentials inside `/var/lib/docker/volumes/production_bitcoin_datadir/_data/bitcoin.conf`. If that is true, how do I run the app through docker on production? 

I saw that the docker-compose.yml file inside BTCPayServer.Tests directory is set to `regtest` and not `mainnet` i.e `BITCOIN_NETWORK: regtest`, which means it is not for production.
